bin/generate-devdocs-selectors-index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const doctrine = require( 'doctrine' );
+const { camelCase, forEach } = require( 'lodash' );
+
+const REGEXP_DOCBLOCKS = /\/\*\* *\n( *\*.*\n)* *\*\//g;
+const SELECTORS_DIR = 'client/state/selectors';
+const DEST_FILE = 'build/devdocs-selectors-index.json';
+
+// Omit index, system files, and subdirectories
+const files = fs
+	.readdirSync( SELECTORS_DIR )
+	.filter( ( file ) => file !== 'index.js' && file.endsWith( '.js' ) );
+
+const selectors = files.map( ( file ) => {
+	const selector = { name: camelCase( path.basename( file, '.js' ) ) };
+
+	const contents = fs.readFileSync( path.join( SELECTORS_DIR, file ), 'utf8' );
+	forEach( contents.match( REGEXP_DOCBLOCKS ), ( docblock ) => {
+		const doc = doctrine.parse( docblock, { unwrap: true } );
+		if ( doc.tags.length > 0 ) {
+			Object.assign( selector, doc );
+			return false;
+		}
+	} );
+
+	return selector;
+} );
+
+selectors.sort( ( a, b ) => a.name > b.name );
+
+fs.writeFileSync( DEST_FILE, JSON.stringify( selectors, null, 2 ) );
```

Comment:
This is failing for me, because at the time we're calling this script, the `/build` directory doesn't exist:

![](https://cldup.com/uNYi-VQCd7.png)

Perhaps before calling the `fs.writeFileSync` we should do a `mkdirp.sync( 'build' );` like we do in other scripts?

Reply:
Fixed :+1: